kafka-connect-redis/src/main/scala/com/datamountaineer/streamreactor/connect/redis/sink/config/RedisSinkSettings.scala

```diff
@@ -25,41 +25,59 @@ import org.apache.kafka.common.config.ConfigException
 import scala.collection.JavaConversions._
 import scala.collection.JavaConverters._
 
+// Redis connection details: host, port, password
+case class RedisConnectionInfo(host: String, port: Int, password: Option[String])
 
-/**
-  * Holds the Redis Sink settings
-  */
-case class RedisSinkSettings(connection: RedisConnectionInfo,
-                             rowKeyModeMap : Map[String, StringKeyBuilder],
-                             routes: List[Config],
-                             fields: Map[String, Map[String, String]],
-                             ignoreFields : Map[String, Set[String]],
-                             errorPolicy : ErrorPolicy = new ThrowErrorPolicy,
-                             taskRetries : Int = RedisSinkConfig.NBR_OF_RETIRES_DEFAULT)
+// Sink settings of each Redis KCQL statement
+case class RedisKCQLSetting(topic: String,
+                            kcqlConfig: Config,
+                            builder: StringKeyBuilder,
+                            fieldsAndAliases: Map[String, String],
+                            ignoredFields: Set[String])
 
+// All the settings of the running connector
+case class RedisSinkSettings(connectionInfo: RedisConnectionInfo,
+                             allKCQLSettings: Set[RedisKCQLSetting],
```

Comment:
why can't it be just kcqlSettings? 'all' can be dropped. it is implied it is all we have in the setting no?